packages/iris-grid/src/IrisGridUtils.test.ts

```diff
@@ -680,3 +667,131 @@ describe('dehydration methods', () => {
     ).toBe(true);
   });
 });
+
+describe('hydration methods', () => {
+  it.each([
+    [
+      'hydrateIrisGridPanelStateV1',
+      {
+        isSelectingPartition: false,
+        partition: null,
+        partitionColumn: 'INVALID',
+        advancedSettings: [],
+      },
+    ],
+    [
+      'hydrateIrisGridPanelStateV2',
+      {
+        isSelectingPartition: false,
+        partitions: [null],
+        partitionColumns: ['INVALID'],
+        advancedSettings: [],
+      },
+    ],
+  ])('%s invalid column error', (_label, panelState) => {
+    expect(() =>
+      IrisGridUtils.hydrateIrisGridPanelState(
+        irisGridTestUtils.makeModel(
+          irisGridTestUtils.makeTable({
+            columns: irisGridTestUtils.makeColumns(5, 'name_'),
+          })
+        ),
+        panelState
+      )
+    ).toThrow('Invalid partition column INVALID');
+  });
+
+  it.each([
+    [
+      'hydrateIrisGridPanelStateV1 null partition',
+      {
+        isSelectingPartition: false,
+        partition: null,
+        partitionColumn: 'name_0',
+        advancedSettings: [],
+      },
+    ],
+    [
+      'hydrateIrisGridPanelStateV1 unselected partition',
+      {
+        isSelectingPartition: false,
+        partition: '',
+        partitionColumn: 'name_0',
+        advancedSettings: [],
+      },
+    ],
+    [
+      'hydrateIrisGridPanelStateV1 one selected partition',
+      {
+        isSelectingPartition: true,
+        partition: '',
+        partitionColumn: 'name_0',
+        advancedSettings: [],
+      },
```

Comment:
These tests don't seem right. A selected partition should have a value for partition,. and the "unselected partition" test - are we testing the empty string? Or `isSelectingPartition`?

Reply:
I believe that the first test is testing for `isSelectingPartition` being `false` (in which case the value of `partition`) shouldn't matter and the second test is just testing for selecting a partition normally. `partition` should be `null` to signify a partition without a value so the second test is just testing for the empty string. I can use a non-empty string to make the test clearer.